spec/registry.md

```diff
@@ -619,6 +619,23 @@ All other values produce an _Invalid Expression_ error.
 > or the type `com.ibm.icu.util.CurrencyAmount` can be used to set the currency and related
 > options (such as the number of fraction digits).
 
+### Non-Negative Integer Options
+
+Some _options_ of number _functions_ are defined to take a "non-negative integer".
+Implementations of number _functions_ use these _options_ to control aspects of numeric display
+such as the number of fraction, integer, or significant digits.
+
+A "non-negative integer" is an _option_ value that the _function_ evaluates as an integer 
+greater than or equal to zero.
+Implementations MAY define an upper limit on the resolved value 
+of a non-negative integer option consistent with that implementation's practical limits.
+
+The implementation MAY accept any implementation-defined type as the value.
+Implementations MUST accept a _literal_ as the value.
```

Comment:
As @eemeli said, this is confusing. Functions aren't passed a data model node, and a _literal_ here could only mean that. 

My suggestion is to define this similarly to how the operand of `:string` is defined:

> The operand of :string is either any implementation-defined type that is a string or for which conversion to a string is supported

and say that a (thing that can be the operand of `:string`) MUST be accepted if it matches `non-negative` in the ABNF.

Perhaps it would be helpful to have a section at the beginning of the registry spec defining the set of implementation-defined types -- that is, it has to at least include a string type, and then, we could have a name for the set of things that can be converted to strings. From there, it would be possible to give a name to the set of things that can be converted to numbers.